app/models/transformation_mapping.rb

```diff
@@ -6,4 +6,72 @@ class TransformationMapping < ApplicationRecord
   def destination(source)
     transformation_mapping_items.find_by(:source => source).try(:destination)
   end
+
+  def validate_vms(csv_list = nil)
+    csv_list.present? ? identify_vms(csv_list) : select_vms
+  end
+
+  private
+
+  def select_vms
+    # TBD
+  end
+
+  def identify_vms(csv_list)
+    valid_list = []
+    invalid_list = []
+    conflict_list = []
+
+    csv_list.each do |row|
+      vm_name = row['Name']
+
+      conditions = {:name => vm_name}
+
+      # construct conditions or selections based on MIQ exported csv columns
+      conditions[:uid_ems] = row['UID'] if row['UID'].present?
+      vms = Vm.where(conditions)
```

Comment:
Instead of building up a conditions Hash, prefer building up queries.

```ruby
query = Vm.where(:name => vm_name)
query = query.where(:uid_ems => row['UID']) if row['UID'].present?
query = query.where(:host => {:name => row['Host']}) if row['Host'].present?

vms = query.to_a
```